Axum example doesn't compile

Cloned main and tried to compile the axum example but was met with (r1.65.0):

```rust
error[E0433]: failed to resolve: could not find `__async_trait` in `entrait`
  --> examples/axum/src/main.rs:14:36
   |
14 |     #[entrait(pub GetFoo, no_deps, box_future, mock_api=GetFooMock)]
   |                                    ^^^^^^^^^^ could not find `__async_trait` in `entrait`

error[E0706]: functions in traits cannot be declared `async`
  --> examples/axum/src/main.rs:14:5
   |
14 |     #[entrait(pub GetFoo, no_deps, box_future, mock_api=GetFooMock)]
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
15 |     async fn get_foo() -> Foo {
   |     ----- `async` because of this
   |
   = note: `async` trait functions are not currently supported
   = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait
   = note: see issue #91611 <https://github.com/rust-lang/rust/issues/91611> for more information
   = note: this error originates in the attribute macro `entrait` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0706]: functions in traits cannot be declared `async`
  --> examples/axum/src/main.rs:15:5
   |
15 |     async fn get_foo() -> Foo {
   |     -----^^^^^^^^^^^^^^^^^^^^
   |     |
   |     `async` because of this
   |
   = note: `async` trait functions are not currently supported
   = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait
   = note: see issue #91611 <https://github.com/rust-lang/rust/issues/91611> for more information

error[E0277]: the trait bound `fn(Extension<A>) -> impl Future<Output = Json<Foo>> {Routes::<A>::get_foo}: Handler<_, _>` is not satisfied
   --> examples/axum/src/main.rs:36:51
    |
36  |             axum::Router::new().route("/foo", get(Self::get_foo))
    |                                               --- ^^^^^^^^^^^^^ the trait `Handler<_, _>` is not implemented for fn item `fn(Extension<A>) -> impl Future<Output = Json<Foo>> {Routes::<A>::get_foo}`
    |                                               |
    |                                               required by a bound introduced by this call
    |
    = help: the trait `Handler<T, ReqBody>` is implemented for `Layered<S, T>`
note: required by a bound in `axum::routing::get`
   --> /home/dkolsoi/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.5.17/src/routing/method_routing.rs:397:1
    |
397 | top_level_handler_fn!(get, GET);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `axum::routing::get`
    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
```

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -22,6 +22,8 @@ jobs:
         run: cargo test --doc --features "unimock use-boxed-futures"
       - name: Clippy
         run: cargo clippy --features "unimock use-boxed-futures" -- -D warnings
+      - name: Build examples
+        run: cargo build --all
 
   test-nightly:
     runs-on: ubuntu-latest

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Unreleased
+### Fixed
+- Make examples compile out of the box, by selecting the correct features.
 
 ## [0.5.0] - 2022-12-10
 ### Changed

--- a/examples/async-graphql/Cargo.toml
+++ b/examples/async-graphql/Cargo.toml
@@ -5,7 +5,7 @@ edition = "2021"
 publish = false
 
 [dependencies]
-entrait = { path = "../../", features = ["unimock"] }
+entrait = { path = "../../", features = ["unimock", "boxed-futures"] }
 async-graphql = "4"
 tokio = { version = "1", features = ["full"] }
 async-trait = "0.1"

--- a/examples/axum/Cargo.toml
+++ b/examples/axum/Cargo.toml
@@ -5,7 +5,7 @@ edition = "2021"
 publish = false
 
 [dependencies]
-entrait = { path = "../../", features = ["unimock"] }
+entrait = { path = "../../", features = ["unimock", "boxed-futures"] }
 axum = "0.5"
 tokio = { version = "1", features = ["full"] }
 serde = { version = "1", features = ["derive"] }

--- a/tests/it/delegation_modes.rs
+++ b/tests/it/delegation_modes.rs
@@ -45,19 +45,12 @@ mod borrow_dyn_use_boxed_futures {
     use async_trait::*;
     use entrait::*;
 
-    #[cfg(feature = "use-boxed-futures")]
-    #[entrait(Foo)]
-    async fn foo(deps: &impl Bar) {
-        deps.bar().await;
-    }
-
-    #[cfg(not(feature = "use-boxed-futures"))]
     #[entrait(Foo, box_future)]
     async fn foo(deps: &impl Bar) {
         deps.bar().await;
     }
 
-    #[entrait(delegate_by = Borrow)]
+    #[entrait(delegate_by = Borrow, box_future)]
     #[async_trait]
     trait Bar: Sync + 'static {
         async fn bar(&self);

